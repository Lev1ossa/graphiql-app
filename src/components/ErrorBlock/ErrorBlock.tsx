import { Refresh, WarningRounded } from '@mui/icons-material'
import { Button, Container, Stack, Typography } from '@mui/material'
import { useNavigate } from 'react-router-dom'

export default function ErrorBlock() {
  const navigate = useNavigate()
  const handleRefreshButtonClick = () => navigate(0)
  return (
    <Container
      component="main"
      maxWidth="md"
      sx={{
        minHeight: 'inherit',
        marginBlock: 2,
        flex: 1,
        display: 'grid',
        placeContent: 'center',
      }}
    >
      <Stack gap="3em" textAlign="center" alignItems="center">
        <Typography variant="h2" fontSize="2.75rem">
          Something went wrong
        </Typography>
        <WarningRounded
          color="error"
          sx={{
            height: '70px',
            width: 'auto',
            aspectRatio: 1,
          }}
        />
        <Button
          startIcon={<Refresh />}
          variant="contained"
          onClick={handleRefreshButtonClick}
        >
          Reload page
        </Button>
      </Stack>
    </Container>
  )
}
