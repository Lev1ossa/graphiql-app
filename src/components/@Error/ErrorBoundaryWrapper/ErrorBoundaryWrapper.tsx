import { Footer } from '@/components/Footer'
import { Header } from '@/components/Header'
import { Stack } from '@mui/material'
import { ErrorBoundary } from '../ErrorBoundary'
import { ErrorBlock } from './ErrorBlock'

export default function ErrorBoundaryWrapper(props: {
  errorHeaderSlot?: React.ReactNode
  children: React.ReactNode
}) {
  const { children, errorHeaderSlot: headerSlot = <Header /> } = props
  const fallback = (
    <Stack minHeight="inherit">
      {headerSlot}
      <ErrorBlock />
      <Footer />
    </Stack>
  )
  return <ErrorBoundary fallback={fallback}>{children}</ErrorBoundary>
}
