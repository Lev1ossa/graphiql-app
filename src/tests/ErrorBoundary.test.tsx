import { ErrorBoundaryWrapper } from '@/components/@Error/ErrorBoundaryWrapper'
import { ErrorBlock } from '@/components/@Error/ErrorBoundaryWrapper/ErrorBlock'
import { expect, test } from 'vitest'
import renderInContext from './renderInContext'

test('Correct ErrorBoundaryWrapper snapshot', () => {
  const { baseElement } = renderInContext(
    <ErrorBoundaryWrapper>{null}</ErrorBoundaryWrapper>,
  )
  expect(baseElement).toMatchSnapshot()
})

test('Correct ErrorBlock snapshot', () => {
  const { baseElement } = renderInContext(<ErrorBlock />)
  expect(baseElement).toMatchSnapshot()
})
