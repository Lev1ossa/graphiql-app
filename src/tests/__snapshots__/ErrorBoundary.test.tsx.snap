// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Correct ErrorBlock snapshot 1`] = `
<body>
  <div>
    <main
      class="MuiContainer-root MuiContainer-maxWidthMd css-x55gft-MuiContainer-root"
    >
      <div
        class="MuiStack-root css-120w1kz-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1k61bmk-MuiTypography-root"
        >
          Something went wrong
        </h2>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-7pb3oa-MuiSvgIcon-root"
          data-testid="WarningRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4.47 21h15.06c1.54 0 2.5-1.67 1.73-3L13.73 4.99c-.77-1.33-2.69-1.33-3.46 0L2.74 18c-.77 1.33.19 3 1.73 3zM12 14c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1s1 .45 1 1v2c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
          />
        </svg>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RefreshIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </span>
          Reload page
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </main>
  </div>
</body>
`;

exports[`Correct ErrorBoundaryWrapper snapshot 1`] = `
<body>
  <div />
</body>
`;
